src/app/charts/chart.component.ts

```diff
@@ -79,15 +100,22 @@ export class ChartComponent implements OnChanges {
             const chartQuery = data[1].url;
             delete data[1].url; // apart from URL, returned data is raw query response
             this.rawChartData = data[1];
-            this.chartData = this.chartService.convertChartData(data);
+            this.processedData = this.chartService.convertChartData(data);
+            this.chartData = _.cloneDeep(this.processedData);
 
             this.curlCommand = `curl -i "${chartQuery}" -H "Authorization: Token ` +
                                `${this.authService.getToken()}"`;
         });
     }
 
-    onSettingsToggleClicked() {
-        this.chart.showSettings = !this.chart.showSettings;
+    sliceChartData() {
+        this.chartData = _.cloneDeep(this.processedData); // to trigger change detection
+        this.chartData[0]['data'] = _.filter(this.chartData[0]['data'], obj => {
```

Comment:
Lodash filter is unnecessary here, `Array.prototype.filter()` has full support and will be transpiled/polyfilled correctly by the build process:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter